File: Data Structures & Algorithms/maximum-subarray/submission-3.java
class Solution {
    public int maxSubArray(int[] nums) {
        if(nums.length == 1){
            return nums[0];
        }
        int total = nums[0];
        int max = nums[0];
        for(int i = 1;i<nums.length;i++){
            if(total < 0){
                total = 0;
            }
            total+= nums[i];
            if(total > max){
                max = total;

            }

        }
        return max;
    }
}

File: Data Structures & Algorithms/max-water-container/submission-3.cpp
class Solution {
public:
    int maxArea(vector<int>& heights) {
        int left = 0;
        int right = heights.size()-1;
        int most = 0;
        while(left<right){
            int volume = (right-left)*std::min(heights[left], heights[right]);
            most = std::max(most, volume);
            if(heights[left]<=heights[right]){
                left++;
            }
            else{
                right--;
            }
        }
        return most;
    }
};
